packages/seacas/libraries/exodus/src/ex_put_field_metadata.c

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright(C) 2022, 2023, 2024 National Technology & Engineering Solutions
+ * of Sandia, LLC (NTESS).  Under the terms of Contract DE-NA0003525 with
+ * NTESS, the U.S. Government retains certain rights in this software.
+ *
+ * See packages/seacas/LICENSE for details
+ */
+
+#include "exodusII.h"     // for ex_err, etc
+#include "exodusII_int.h" // for EX_FATAL, etc
+
+static int exi_print_type_error(int status, const char *name, const char *type,
+                                const char *attribute, int exoid, const char *func)
+{
+  char errmsg[MAX_ERR_LENGTH];
+  snprintf(errmsg, MAX_ERR_LENGTH, "ERROR: failed to store '%s' for %s '%s' in file id %d",
+           attribute, type, name, exoid);
+  ex_err_fn(exoid, func, errmsg, status);
+  return EX_FATAL;
+}
+
+static int exi_print_attribute_error(int status, const char *name, const char *attribute,
+                                     ex_entity_type entity_type, ex_entity_id entity_id, int exoid,
+                                     const char *func)
+{
+  char errmsg[MAX_ERR_LENGTH];
+  snprintf(errmsg, MAX_ERR_LENGTH,
+           "ERROR: failed to store field metadata '%s' for field '%s' on %s with id %" PRId64
+           " in file id %d",
+           attribute, name, ex_name_of_object(entity_type), entity_id, exoid);
+  ex_err_fn(exoid, func, errmsg, status);
+  return EX_FATAL;
+}
+
+int ex_put_field_metadata(int exoid, const ex_field field)
+{
+  /*
+   * The attribute name is `Field@{name}@type`
+   *
+   * Field `nesting` is calculated as size of `type` field
+   * The `type` field is a sequence of integers which are the values of the `ex_field_type` enum.
+   * NOTE: For backward compatibility, we can only add new entries to this enum at the end.
+   *
+   * If size of `component_separator` == 0 then the default '_' separator used by all component
+   * levels If size of `component_separator` == 1 then that separator used by all component levels
+   * Else the size must equal nesting and it specifies a potentially different separator for each
+   * level.
+   */
+#if 0
```

Comment:
Perhaps this line should be something like '#if PRINT_DEBUG_STUFF' instead of '#if 0'